### PR TITLE
Implements Redux Best Practices as of Aug. 2020.

### PR DESCRIPTION
**Pull request by sandroroux**
Closes catalpainternational/asteroid#209.

This PR implements some Redux [best practices](https://redux.js.org/style-guide/style-guide) and loosely couples Canoe and Redux.

## Redux Best Practices
This PR Implements:
* [Structure Files As Feature Folders or Ducks](https://redux.js.org/style-guide/style-guide#structure-files-as-feature-folders-or-ducks)
* [Model Actions as Events, Not Setters](https://redux.js.org/style-guide/style-guide#model-actions-as-events-not-setters)
* [Write Meaningful Action Names](https://redux.js.org/style-guide/style-guide#write-meaningful-action-names)
* [Write Action Types as `domain/eventName`](https://redux.js.org/style-guide/style-guide#write-action-types-as-domaineventname)
* [Use Action Creators](https://redux.js.org/style-guide/style-guide#use-action-creators)

## Loose coupling between Canoe and Redux
I  moved all Store getters and setters to `Interface.js`. `Interface.js` executes Dependency Inversion.

I removed all direct Redux store references from `ServiceWorkerManagement.js`, `Translation.js`, and `app.js`.

### CODE REVIEW

File: src/js/Translation.js [hunk 1]
```diff
@@ -30,16 +31,16 @@ function setLocale(locale) {
     i18n.setLocale(locale);
 }
 
-function storeDispatch() {
-    const newStoreState = store.getState();
-    if (newStoreState.language !== previousStoreState.language) {
-        setLocale(newStoreState.language);
+function checkIfLanguageChanged() {
```
Comment (PeteCoward) [inline src/js/Translation.js:34]:
```suggestion
function updateLocaleIfLanguageChanged() {
```

File: src/js/Translation.js [hunk 2]
```diff
@@ -30,16 +31,16 @@ function setLocale(locale) {
...
+    previousLanguageState = newLanguageState;
 }
 
 const userLanguage = getLanguage();
 setLocale(userLanguage);
 
-let previousStoreState = store.getState();
-const unsubscribe = store.subscribe(storeDispatch);
+let previousLanguageState = userLanguage;
+const unsubscribe = subscribeToStore(checkIfLanguageChanged);
```
Comment (PeteCoward) [inline src/js/Translation.js:46]:
```suggestion
const unsubscribe = subscribeToStore(updateLocaleIfLanguageChanged);
```

File: src/riot/Components/DataDownload.riot.html [hunk 3]
```diff
@@ -48,14 +47,12 @@
...
                 this.state.completionPoller = window.setInterval(updateApi, EVERY_FIVE_MINUTES);
 
-                storeSiteDownloadedIs(true);
                 dispatchLoggedInEvent();
             },
 
             async clearAppData() {
                 window.clearInterval(this.state.completionPoller);
                 window.clearInterval(this.state.pagesPoller);
-                storeSiteDownloadedIs(false);
```
Comment (PeteCoward) [inline src/riot/Components/DataDownload.riot.html:58]: so glad to kill this yoda named method :)

**Comment by PeteCoward** [comment]
For me the biggest aim with this redux tech is to get to the point where we can use it for testing.
Where we can control the app state by calling redux actions, without any of the real dependencies ( apis ) being present.

File: src/riot/Components/DataDownload.riot.html (hunk 3 above)
Comment (sandroroux) [reply src/riot/Components/DataDownload.riot.html:58]: @PeteCoward, I wonder whether we can implement other flags as methods that inspect Redux data.

**Comment by PeteCoward** [comment]
I think we should get this into an RC-1.3 as a base so it is well tested.

**Comment by sandroroux** [comment]
I pointed this at RC-1.3.

**Review by PeteCoward — APPROVED** [review]
I'l leave it to you to merge as you may have more context than me

**Pull request merged**